src/epics/detectPageVisibilityObservable.js

```diff
@@ -0,0 +1,36 @@
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/observable/merge';
+import 'rxjs/add/observable/of';
+
+export default function createDetectPageVisibilityObservable() {
+
+  let hiddenKeyName, visibilityChangeEventName;
+  if (typeof document.hidden !== "undefined") { // Opera 12.10 and Firefox 18 and later support
+    hiddenKeyName = "hidden";
+    visibilityChangeEventName = "visibilitychange";
+  } else if (typeof document.msHidden !== "undefined") {
+    hiddenKeyName = "msHidden";
+    visibilityChangeEventName = "msvisibilitychange";
+  } else if (typeof document.webkitHidden !== "undefined") {
+    hiddenKeyName = "webkitHidden";
+    visibilityChangeEventName = "webkitvisibilitychange";
+  }
+
+  const detectPageVisibilityObservable = Observable.merge(
+    Observable.fromEvent(document, visibilityChangeEventName)
+      .map(() => {
+        if (document[hiddenKeyName]) {
+          return 'online';
+        } else {
+          return 'viewing';
+        }
+      }),
+    Observable.fromEvent(window, 'beforeunload')
+      .do(e => e.returnValue = 'Do you really want to quit?')
+      .map(() => 'offline')
```

Comment:
@Spetastian So it makes sense to use `map` in these cases where there's only 1 value?  Usually I think of `map`ping over an array or the like... but should I be thinking about this in terms of a _stream_ of values?

Reply:
each of which is being mapped to some other value with `.map(...)`?

Reply:
Yes, you can se `.map(...)` in this way:
Map is an operator that transforms every value from a source into another value. For an array the source has a fixed set of values and the map call operates upon each and one of them.

In the case of an Observable stream, the map operator works the same but instead of a fixed set of values it can have an infinite amount of values pushed to the sources at different points in time.

In this specific case the consumer of this observable is not interested in the event object that the `Observabe.fromEvent(...)` emits. The only interesting values are the three states "online", "viewing" and "offline". 

Do you feel like this is a good explanation as to why we are using .map to map to a single value here?